packages/typography/src/Body.tsx

```diff
@@ -2,45 +2,59 @@ import React from 'react';
 import Box from '@leafygreen-ui/box';
 import { css, cx } from '@leafygreen-ui/emotion';
 import { HTMLElementProps } from '@leafygreen-ui/lib';
-import { sharedStyles, typeScale1, typeScale2 } from './styles';
-import { useBaseFontSize } from '@leafygreen-ui/leafygreen-provider';
+import { palette } from '@leafygreen-ui/palette';
+import { baseTypographyStyles, bodyTypeScaleStyles } from './styles';
+import { CommonTypographyProps, Mode } from './types';
+import { useUpdatedBaseFontSize } from '.';
 
 /**
  * Body
  */
 
 type BodyFontWeight = 'regular' | 'medium';
-type BodyProps<T extends keyof JSX.IntrinsicElements> = HTMLElementProps<T> & {
-  /**
-   * font-weight applied to typography element
-   * default: `regular`
-   */
-  weight?: BodyFontWeight;
-  as?: T;
+type BodyProps<T extends keyof JSX.IntrinsicElements> = HTMLElementProps<T> &
+  CommonTypographyProps & {
+    /**
+     * font-weight applied to typography element
+     * default: `regular`
+     */
+    weight?: BodyFontWeight;
+    as?: T;
+  };
+
+const fontWeights: Record<
+  'default' | 'strong',
+  Record<BodyFontWeight, number>
+> = {
+  default: {
+    regular: 400,
+    medium: 500,
+  },
+  strong: {
+    regular: 700,
+    medium: 700,
+  },
+} as const;
+
+const bodyColor: Record<Mode, string> = {
+  [Mode.Light]: css`
+    color: ${palette.black};
+  `,
+  [Mode.Dark]: css`
+    color: ${palette.gray.light2};
+  `,
```

Comment:
Could probably export this as something like `defaultTextColor` since it's re-used often, and just define the colors that break the pattern (like H1, H2 & Link)